c/util.c

```diff
@@ -0,0 +1,27 @@
+#include <stdio.h> /* printf */
+#include <string.h> /* strrchr */
+#include <libgen.h>
+#define _GNU_SOURCE
+
+char* get_file_extension(char* path) {
+    const char ch = '.';    
+    return strrchr(path, ch);
+}
+
+char* get_file_name(char* path) {
+    char* basec = strdup(path);
+    return basename(basec);
+}
+
+char* get_dir_files(char* path) {
```

Comment:
Essa função deveria retornar todos os arquivos de um diretorio. No momento ela retorna o nome do diretorio a partir de um caminho. Acredito que essa função deva existir no entanto o nome dela deve ser mudado.

```suggestion
char* get_dir_path(char* path) {
```